test/ProgressOnderwijsUtils.Tests/Data/CascadedDeleteTest.cs

```diff
@@ -61,6 +62,32 @@ public void CascadedDeleteFollowsAUniqueConstraintBasedForeignKey()
             PAssert.That(() => deletionReport.Select(t => t.Table).SequenceEqual(new[] { "dbo.T2", "dbo.T1" }));
         }
 
+
+        [Fact]
+        public void CascadedDeleteDetectsCycles()
+        {
+            SQL($@"
+                create table T1 ( A int not null unique, B int);
+                create table T2 ( B int not null unique, A int references T1 (A) );
+                alter table T1 add constraint T1FK foreign key (B) references T2(B);
+
+                insert into T1 values (1,null), (2,null), (3, null);
+                insert into T2 values (111,1), (333,3);
+                
+                update T1 set B = 111 where A = 1;
+                update T1 set B = 333 where A = 3;
+
+            ").ExecuteNonQuery(Connection);
+
+            var initialDependentValues = SQL($"select B from T2").ReadPlain<int>(Connection);
+
+            PAssert.That(() => initialDependentValues.SetEqual(new[] { 111, 333 }));
+
+            var db = DatabaseDescription.LoadFromSchemaTables(Connection);
+            var ex =Assert.ThrowsAny<Exception>(()=> CascadedDelete.RecursivelyDelete(Connection, db.GetTableByName("dbo.T1"), false, null, null, "A", AId.One, AId.Two));
+            PAssert.That(() => ex.Message.Contains("dbo.T2->dbo.T1->dbo.T2->dbo.T1"));

```

Comment:
Dus netto gaat het hierom; het pad naar de cycle wordt veel sneller gedetecteerd, maar ook met een veel betere foutmelding.  de root (T2) komt hier 2 keer in voor omdat ie anders met zijn keys omgaat, de echte gedetecteerde cycle is  T1->T2->T1.

Reply:
(oh, en de cycle-check is echt op de data gebaseerd, dus tabel-cycles zijn prima, als de rows maar geen cycles maken).